crates/subspace-farmer/src/bin/subspace-farmer/commands/farm/fixed_provider_storage.rs

```diff
@@ -0,0 +1,91 @@
+use crate::commands::farm::ReadersAndPieces;
+use parking_lot::Mutex;
+use std::borrow::Cow;
+use std::sync::Arc;
+use subspace_networking::libp2p::kad::record::Key;
+use subspace_networking::libp2p::kad::ProviderRecord;
+use subspace_networking::libp2p::PeerId;
+use subspace_networking::utils::multihash::MultihashCode;
+use subspace_networking::{deconstruct_record_key, ProviderStorage, ToMultihash};
+
+pub(crate) struct FixedProviderStorage {
```

Comment:
Why does `FixedProviderStorage` still exist? I though we'd get rid of it, but now instead of simplifying things we're making it even more complex.